solar-system/src/org/gearvrf/solarsystem/LightMapManager.java

```diff
@@ -0,0 +1,181 @@
+        }
+
+        public GVRCustomMaterialShaderId getShaderId() {
+            return mShaderId;
+        }
+    }
+
+    protected class LightMapInfo {
+
+        public class LightMapInfoObject {
+            public String name;
+            public float offsetX;
+            public float offsetY;
+            public float scaleX;
+            public float scaleY;
+        }
+
+        private List<LightMapInfoObject> mMapInfoList;
+
+        public LightMapInfo(GVRContext gvrContext, int lighMapInfoId) {
+            mMapInfoList = new ArrayList<LightMapInfoObject>();
+
+            loadLightMapInfo(gvrContext, lighMapInfoId);
+        }
+
+        private void loadLightMapInfo(GVRContext gvrContext, int lighMapInfoId) {
+            try {
+                InputStream is = gvrContext.getActivity().getResources().openRawResource(lighMapInfoId);
+                int size = is.available();
+                byte[] buffer = new byte[size];
+
+                is.read(buffer);
+
+                is.close();
+
+                loadLightMapInfo(gvrContext, new JSONArray(new String(buffer, "UTF-8")));
+            } catch (JSONException je) {
+                je.printStackTrace();
+            } catch (IOException ex) {
+                ex.printStackTrace();
+            }
+        }
+
+        private void loadLightMapInfo(GVRContext gvrContext, JSONArray jsonInfo) throws JSONException {
+            int length = jsonInfo.length();
+
+            for (int i = 0; i < length; i++) {
+                if (jsonInfo.isNull(i))
+                    continue;
+
+                JSONObject jsonObj = jsonInfo.getJSONObject(i);
+                LightMapInfoObject lmiObj = new LightMapInfoObject();
+
+                lmiObj.name = jsonObj.getString("name");
+                lmiObj.offsetX = (float) jsonObj.getDouble("offset.x");
+                lmiObj.offsetY = (float) jsonObj.getDouble("offset.y");
+                lmiObj.scaleX = (float) jsonObj.getDouble("scale.x");
+                lmiObj.scaleY = (float) jsonObj.getDouble("scale.y");
+
+                mMapInfoList.add(lmiObj);
+            }
+        }
+
+        public int lenght() {
+            return mMapInfoList.size();
+        }
+
+        public LightMapInfoObject get(int i) {
+            return mMapInfoList.get(i);
+        }
+    }
+
+    private GVRTexture mLightMapTexture;
+    private LightMapInfo mLighMapInfo;
+    private LightMapShader mLightMapShader;
+
+    LightMapManager(GVRContext gvrContext, int lighMapTextureId, int lighMapInfoId) {
+        loadLightMapTexture(gvrContext, lighMapTextureId);
+        mLighMapInfo = new LightMapInfo(gvrContext, lighMapInfoId);
+        mLightMapShader = new LightMapShader(gvrContext);
+    }
+
+    public void applyLightMap(GVRContext gvrContext, GVRScene scene) {
+        int lenghtLightmapInfo = mLighMapInfo.lenght();
+
+        for (int i = 0; i < lenghtLightmapInfo; i++) {
```

Comment:
It would be better to provide a getter for the `List<LightMapInfoObject>`. Then, the user can use `for (LightMapInfoObject lmiObject : mapInfoList)` to iterate the list. This is more robust and less error-prone.